workflow/controller/operator.go

```diff
@@ -191,14 +191,30 @@ func (woc *wfOperationCtx) operate() {
 
 	woc.log.Infof("Processing workflow")
 
-	// Load the WorkflowSpec for execution
-	execTmplRef, execArgs, err := woc.loadExecutionSpec()
+	// Loading the Execute workflow spec for execution
+	// ExecWF is a runtime execution spec which merged from Wf, WFT and Wfdefault
+	err := woc.loadExecWFFromWfDefaultOrWFTRef()
 	if err != nil {
-		woc.log.WithError(err).Errorf("Unable to get Workflow Template Reference for workflow")
+		woc.log.Errorf("loading WorkflowDefault or WorkflowTemplateRef failed: %v", err)
 		woc.markWorkflowError(err, true)
 		return
 	}
 
+	// Workflow Level Synchronization lock
+	if woc.execWf.Spec.Synchronization != nil {
+		acquired, wfUpdate, _, err := woc.controller.syncManager.TryAcquire(woc.wf, "", woc.execWf.Spec.Synchronization)
```

Comment:
Why are we suddenly ignoring `msg` here?

Reply:
Good catch, Previous this code was in controller.go that's why message was ignored. But here we can update the wf object.